tasks/discover-granules/tests/index.js

```diff
@@ -137,69 +117,66 @@ test('discover granules using HTTP', async (t) => {
-  await s3().createBucket({ Bucket: event.config.bucket }).promise();
 
   try {
-    const output = await discoverGranules(event);
-    await validateOutput(t, output);
-    t.is(output.granules.length, 3);
-    t.is(output.granules[0].files.length, 2);
-    t.truthy(['data', 'metadata'].includes(output.granules[0].files[0].type));
-  } catch (err) {
-    if (err.message === 'Connection Refused') {
+    await assertDiscoveredGranules(t, await discoverGranules(event));
+  } catch (e) {
+    if (e.message === 'Connection Refused') {
       t.pass('Ignoring this test. Remote host seems to be down.');
-    } else throw err;
-  } finally {
-    // Clean up
-    await recursivelyDeleteS3Bucket(event.config.bucket);
+    } else throw e;
   }
 });
 
-test('discover granules using S3', async (t) => {
-  const internalBucketName = randomString();
-  const sourceBucketName = randomString();
-  const providerPath = randomString();
-
-  // Create providerPathDirectory and internal bucket
-  await Promise.all([
-    s3().createBucket({ Bucket: internalBucketName }).promise(),
-    s3().createBucket({ Bucket: sourceBucketName }).promise()
-  ]);
-
+const discoverGranulesUsingS3 = (configure) => async (t) => {
+  const { event, event: { config } } = t.context;
   // State sample files
   const files = [
     'granule-1.nc', 'granule-1.nc.md5',
     'granule-2.nc', 'granule-2.nc.md5',
     'granule-3.nc', 'granule-3.nc.md5'
   ];
-  await Promise.all(files.map((file) =>
-    s3().putObject({
-      Bucket: sourceBucketName,
-      Key: `${providerPath}/${file}`,
-      Body: `This is ${file}`
-    }).promise()));
 
-  const { event } = t.context;
-  event.config.collection.provider_path = providerPath;
-  event.config.provider = {
-    id: 'MODAPS',
-    protocol: 's3',
-    host: sourceBucketName
-  };
+  config.sourceBucketName = randomString();
+  config.collection.provider_path = randomString();
 
-  await validateConfig(t, event.config);
+  configure(config);
+
+  await validateConfig(t, config);
+  await s3().createBucket({ Bucket: config.sourceBucketName }).promise();
 
   try {
-    const output = await discoverGranules(event);
-    await validateOutput(t, output);
-    t.is(output.granules.length, 3);
-    t.is(output.granules[0].files.length, 2);
-    t.truthy(['data', 'metadata'].includes(output.granules[0].files[0].type));
+    await Promise.all(files.map((file) =>
+      s3().putObject({
+        Bucket: config.sourceBucketName,
+        Key: `${config.collection.provider_path}/${file}`,
+        Body: `This is ${file}`
+      }).promise()));
+    await assertDiscoveredGranules(t, await discoverGranules(event));
   } finally {
     // Clean up
-    await Promise.all([
-      recursivelyDeleteS3Bucket(internalBucketName),
-      recursivelyDeleteS3Bucket(sourceBucketName)
-    ]);
+    await recursivelyDeleteS3Bucket(config.sourceBucketName);
   }
-});
+};
+
+test('discover granules using S3',
+  discoverGranulesUsingS3((config) => {
```

Comment:
is this call missing an invocation? I may be reading things wrong, but it looks to me like `discoverGranulesUsingS3` returns an async function that wants some kind of `t` input.

Reply:
In this case, the call to `discoverGranulesUsingS3` is returning an `async` function that takes a `t` argument, which is what the `test` function expects as its second argument. At first glance it's perhaps not obvious since it's not in the usual explicit syntax we normally use. This makes it convenient to use both for this test and the following test.

Reply:
:doh: thanks for explaining that to me.